### PR TITLE
test: Sort 

### PR DESCRIPTION
**Pull request by SilviaZeta**
## Description

- refactor `searchAndSort` folder 
- move sort functions outside of `ui.tsx` to facilitate unit testing (and code refactor)
- add unit test for `sortMetadata` function

## Dependency changes
No

## Documentation

No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New migrations have been committed
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] If appropriate, I have bumped any version numbers

### CODE REVIEW

File: src/components/AutoAssignDialog.test.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { render, fireEvent, screen, within } from "@testing-library/react";
-import { Metadata } from "../searchAndSort/interfaces";
+import { Metadata } from "../interfaces";
```
Comment (ChasNelson1990) [inline src/components/AutoAssignDialog.test.tsx:2]: why not `"@/interfaces"`?

File: src/components/AutoAssignDialog.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ import { BaseIconButton, BaseTextButton, theme, icons } from "@gliff-ai/style";
 import { tooltips } from "./Tooltips";
 import { Profile } from "./interfaces";
 import { kCombinations, shuffle } from "../helpers";
-import { Metadata, MetaItem } from "../searchAndSort/interfaces";
+import { Metadata, MetaItem } from "../interfaces";
```
Comment (ChasNelson1990) [inline src/components/AutoAssignDialog.tsx:21]: why not `"@/interfaces"`?

File: src/helpers.test.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { Metadata } from "./interfaces";
```
Comment (ChasNelson1990) [inline src/helpers.test.ts:1]: why not `"@/interfaces"`?

File: src/search/SearchBar.tsx [hunk 4]
```diff
@@ -7,7 +7,7 @@ import Autocomplete from "@material-ui/lab/Autocomplete";
 import { BaseIconButton, theme } from "@gliff-ai/style";
 import { metadataNameMap } from "@/MetadataDrawer";
 import { tooltips } from "@/components/Tooltips";
-import { Metadata, MetaItem, Filter } from "./interfaces";
+import { Metadata, MetaItem, Filter } from "../interfaces";
```
Comment (ChasNelson1990) [inline src/search/SearchBar.tsx:10]: why not `"@/interfaces"`?

**Review by ChasNelson1990 — APPROVED** [review]
I think this probably should have been two PRs - one to refactor, one to add tests

**Comment by SilviaZeta** [comment]
> I think this probably should have been two PRs - one to refactor, one to add tests

I think so too. next time..

**Review by philipjackson — APPROVED** [review]

**Review by cooper667 — APPROVED** [review]

**Comment by cooper667** [comment]
I think we should allow `any` and `console` in our lint rules for test files?

**Pull request merged**